File: library/src/main/java/io/ashdavies/rx/rxtasks/TaskListenerFactory.java
package io.ashdavies.rx.rxtasks;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;

interface TaskListenerFactory<Result, Emitter> {

  OnSuccessListener<Result> createOnSuccessListener(Emitter emitter);

  OnFailureListener createOnFailureListener(Emitter emitter);
}

File: library/src/main/kotlin/io/ashdavies/rx/rxtasks/TaskListenerFactory.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener

internal interface TaskListenerFactory<Result, Emitter> {

  fun createOnSuccessListener(emitter: Emitter): OnSuccessListener<Result>

  fun createOnFailureListener(emitter: Emitter): OnFailureListener
}
